src/events/components/DetailView/PublicAttendees/index.tsx

```diff
@@ -1,29 +1,46 @@
-import React, { FC, useEffect } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 
 import { useDispatch, useSelector } from 'core/redux/hooks';
 import { selectPublicAttendeesForEventId } from 'events/selectors/publicAttendee';
 import { fetchPublicAttendeesByEventId } from 'events/slices/publicAttendees';
 
 import { Attendee } from './Attendee';
 import style from './PublicAttendees.less';
+import { IPublicAttendee } from 'events/models/Attendee';
+import { Button, Modal } from '@dotkomonline/design-system';
 
 interface IProps {
   eventId: number;
+  isAttending: boolean;
 }
 
-export const PublicAttendees: FC<IProps> = ({ eventId }) => {
+const attendeeList = (attendees: IPublicAttendee[]) => (
+  <div className={style.grid}>
+    {attendees.map((attendee, index) => (
+      <Attendee key={attendee.id} attendee={attendee} count={index + 1} />
+    ))}
+  </div>
+);
+
+export const PublicAttendees: FC<IProps> = ({ eventId, isAttending }) => {
   const dispatch = useDispatch();
   const attendees = useSelector(selectPublicAttendeesForEventId(eventId));
+  const [showModal, setShowModal] = useState(false);
+
+  const toggleModal = () => setShowModal(!showModal);
 
   useEffect(() => {
     dispatch(fetchPublicAttendeesByEventId(eventId));
   }, [eventId]);
 
+  if (!isAttending) return null;
```

Comment:
Why was this moved to inside the component?
Would that not result in the fetch function being called even when it shouldn't?

Reply:
This is fair. I moved it inside since it would make the parent file less cluttery. Either then creating another component that does the null check or writing a conditional render, or another soluton.

Reply:
This has now been moved into its own component and logic.